### PR TITLE
OCR support

### PR DESCRIPTION
**Pull request by blitz**
In this PR:
- implementation of `get_screen_text` and `wait_for_text`
- a `retry` function that is like the Perl one
- conversion of the `emacs-daemon` test

### CODE REVIEW

File: nixos/lib/test-driver/test-driver.py [hunk 1]
```diff
@@ -413,6 +429,43 @@ def screenshot(self, filename):
             if ret.returncode != 0:
                 raise Exception('Cannot convert screenshot')
 
+    def get_screen_text(self):
+        if shutil.which("tesseract") is None:
```
Comment (tfc) [inline nixos/lib/test-driver/test-driver.py]:
```suggestion
        if shutil.which('tesseract') is None:
```

File: nixos/lib/test-driver/test-driver.py [hunk 2]
```diff
@@ -413,6 +429,43 @@ def screenshot(self, filename):
...
+            raise Exception('get_screen_text used but enableOCR is false')
+
+        magick_args = """
+        -filter Catrom -density 72 -resample 300
+        -contrast -normalize -despeckle -type grayscale
+        -sharpen 1 -posterize 3 -negate -gamma 100
+        -blur 1x65535
+        """.replace('\n', '')
```
Comment (tfc) [inline nixos/lib/test-driver/test-driver.py]:
```
magick_args = ('-filter Catrom -density 72 -resample 300 '
              + '-contrast -normalize -despeckle -type grayscale '
              + '-sharpen 1 -posterize 3 -negate -gamma 100 '
              + '-blur 1x65535')
```

Comment (blitz) [reply nixos/lib/test-driver/test-driver.py]: Done.

**Comment by blitz** [comment]
@tfc I've updated the PR.

**Pull request merged**